raft-tech/TANF-app
# 3157 - reparse meta additions

## Summary of Changes
Pull request closes #3157 
Create a new `ReparseFileMeta` through model for the m2m relationship between `DataFile` and `ReparseMeta`
* adds `cat4_error_count` for each datafile included in a reparse
* adds `finished_at` for each datafile included in a reparse
* moves `success`, `finished`, and `num_records_created` to be reported for each datafile in a reparse
* modifies the `ReparseMeta` to aggregate `ReparseFileMeta` data via properties, rather than storing data directly on the model

## How to Test

```
cd tdrs-backend && docker compose up --build
cd tdrs-frontend && docker compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Submit some datafiles and create reparse events for them
1. Ensure data reported on `ReparseFileMeta` is accurate. Ensure data reported on `ReparseMeta` is accurate.
1. Deploy `develop`. Repeat the tests on the old code. Run the migrations on the new branch and ensure that the through-models are correctly re-created


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
